e2e_test/batch/catalog/rw_depend.slt.part

```diff
@@ -34,6 +34,7 @@ join rw_relations r1 on d.objid = r1.id
 join rw_relations r2 on d.refobjid = r2.id;
 ----
 idx1  t1
+mv1 idx1
```

Comment:
It seems we will still capture the dependencies of `mv1 -> t1`, if we select the index idx1 instead of t1. I think it is acceptable, but where do we record this dependency? If I create the materialized view by using the index explictly like ` create materialized view mv1 as select idx1.a from idx1 join s1 on idx1.a = s1.a;` then there is no dependency of `mv1 -> t1` cc @yezizp2012

Reply:
The dependency info will be resolved during both binding and planning. If we explicitly specify the index in the mv's definition then the index will be the only referenced object.

Reply:
> It seems we will still capture the dependencies of `mv1 -> t1`, if we select the index idx1 instead of t1. I think it is acceptable

Will this allow the index to be dropped without checking presence of materialized view atop of it, which sounds problematic?

Reply:
No. This above issue means mv1 not only depends on the index idx1, but also depends on the table t1. So the index can't be dropped without dropping the mv first.

Reply:
> If I create the materialized view by using the index explictly like  create materialized view mv1 as select idx1.a from idx1 join s1 on idx1.a = s1.a; then there is no dependency of mv1 -> t1

I do found a issue for this case, the table `t1` can be dropped in restrict mode, which is problematic and should be rejected. I opened a PR to fix it: #23101 .